dotnet/src/Connectors/Connectors.OpenAI/AudioToText/OpenAIAudioToTextService.cs

```diff
@@ -66,6 +67,50 @@ public sealed class OpenAIAudioToTextService : IAudioToTextService
         this._core.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
     }
 
+    /// <summary>
+    /// Creates an instance of the <see cref="OpenAIAudioToTextService"/> for Custom Audio OpenAI API
+    /// </summary>
+    /// <param name="modelId">Model name</param>
+    /// <param name="endpoint">Custom OpenAI Audio Transcription API compatible endpoint</param>
+    /// <param name="apiKey">OpenAI API Key</param>
+    /// <param name="organization">OpenAI Organization Id (usually optional)</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    public OpenAIAudioToTextService(
+        string modelId,
+        Uri endpoint,
+        string? apiKey = default,
+        string? organization = null,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        Uri? internalClientEndpoint = null;
+        var providedEndpoint = endpoint ?? httpClient?.BaseAddress;
+        if (providedEndpoint is not null)
+        {
+            // If the provided endpoint does not have a path specified, updates it to the default Message API Chat Completions endpoint
+            internalClientEndpoint = providedEndpoint.PathAndQuery == "/" ?
+                new Uri(providedEndpoint, "v1/audio/transcriptions")
+                : providedEndpoint;
+        }
+
+        this._core = new(
```

Comment:
VerifyNullOrWhite the `modelId` and VerifyNull the `internalEndpoint`.

Using this constructor both should be required.